[mobile] fix docs page

jankeromnes opened the pull request:
Expandable table of contents in mobile menu (I also include the ToC at the bottom of the mobile Docs page, for convenience):

<img width="837" alt="Screenshot 2019-03-19 at 16 53 04" src="https://user-images.githubusercontent.com/599268/54621350-e7a5cb00-4a67-11e9-8adb-f2b029b7c6dc.png">

---

Docs page on mobile (needs a bit more padding at the top, but otherwise quite readable):

<img width="410" alt="Screenshot 2019-03-19 at 16 54 10" src="https://user-images.githubusercontent.com/599268/54621443-0ad07a80-4a68-11e9-91e3-4402c991b17d.png">

---

Docs page on mobile in landscape mode (a bit more padding left and right -- iPad in portrait looks the same):

<img width="706" alt="Screenshot 2019-03-19 at 16 54 44" src="https://user-images.githubusercontent.com/599268/54621548-3489a180-4a68-11e9-8bee-8d0e5ad41b32.png">

---

iPad in landscape mode is able to show the full page (non-mobile version).

jankeromnes commented:
The up/down arrow is actually the left arrow rotated. It has a strange blue tint.

ChristinFrohne commented:
looks good! 
- the spacing in the menu is better now. But i would still give it more space above the logo. 
- any reason why you left out the other sites like 'contact' or 'about' in the menu? Is it too much for now and we will add them another time? 
- What is the problem with the arrow? Can you not get rid of the blue glow?
- I would use a smaller 'log in' button on the mobile screen
- for the doc screen: give it some more space to the sides and the top

jankeromnes commented:
> looks good!

Thanks!

> the spacing in the menu is better now. But i would still give it more space above the logo.

Ok, I'll try to make the top bar taller to give it more space.

> any reason why you left out the other sites like 'contact' or 'about' in the menu?

I just used the links that were already visible in the non-mobile menu.

- However, I can add an About link (should it also be visible in the non-mobile menu?)
- There is no Contact page. Should I add one?
- Note: The footer has links to Features, Pricing, Docs, Blog, About, Careers, Status, Enterprise. (Should I add a link to Contact there as well?)

> What is the problem with the arrow? Can you not get rid of the blue glow?

Yes the problem is that it's a bit blue. I should probably remove the blue glow with CSS.

> for the doc screen: give it some more space to the sides and the top

I totally agree with more space at the top, however I feel like increasing the padding on the sides is not ideal, because mobile screens are already so narrow (too much padding will make each page very very long).

I'll try to read through the docs on an actual phone to see if we need more left/right padding.

svenefftinge commented:
I first didn't find the menu. Only after reading the comment here I figured it was nested within the main menu. Not a good sign.

The resizing isn't good for other sizes. E.g.:

<img width="748" alt="Screenshot 2019-03-20 at 09 01 30" src="https://user-images.githubusercontent.com/372735/54668268-c6d48880-4aee-11e9-8533-5ddbc7350df4.png">

jankeromnes commented:
> I first didn't find the menu.

Thanks for the feedback. There are two docs menu on mobile: expandable in mobile menu, and at the bottom of each docs page.

Please suggest a better place (or places).

jankeromnes commented:
> The resizing isn't good for other sizes. E.g.:
> 674 x 710

Is that a common window size? I usually test for mobile (375 x 667), mobile landscape (667 x 375), iPad (768 x 1024) and iPad landscape (1024 x 768).

jankeromnes commented:
> Actually it doesn't look right in any size. Did you try using the responsive mode in the dev tools?

Yes, I'm using Firefox Devtools' reponsive design, in which the website looks great (and also on a real device).

I've also tried Chrome Devtools responsive tool, and was able to reproduce your bug. It's because of the Gitpod logo all the way to the right. I'll remove it.

jankeromnes commented:
- [x] rebased
- [x] adjusted mobile menu bar sizing
- [x] added About and Contact links (only on mobile, also currently gives "404 page not found")
- [x] fixed Chrome-only overflow bugs
- [x] increased top padding of Docs content on mobile

jankeromnes commented:
New look in Firefox:

<img width="409" alt="Screenshot 2019-03-20 at 10 35 30" src="https://user-images.githubusercontent.com/599268/54674300-68aea200-4afc-11e9-8f24-3d9d578ffc1a.png">

And in Chrome:

<img width="417" alt="Screenshot 2019-03-20 at 10 38 33" src="https://user-images.githubusercontent.com/599268/54674312-6ea48300-4afc-11e9-8e3c-862ec7fc5cd9.png">

svenefftinge commented:
Cool, looks much better! 👍 

In this size there is no menu:
<img width="800" alt="Screenshot 2019-03-20 at 11 40 15" src="https://user-images.githubusercontent.com/372735/54678244-f3939a80-4b04-11e9-9d03-e53008a253b4.png">

svenefftinge commented:
We should disable the background logos.
<img width="291" alt="Screenshot 2019-03-20 at 11 42 43" src="https://user-images.githubusercontent.com/372735/54678382-54bb6e00-4b05-11e9-81e9-c13f7b72a8ed.png">

svenefftinge commented:
For the menu I like how netlify do it in a separate combo box.

<img width="465" alt="Screenshot 2019-03-20 at 11 44 43" src="https://user-images.githubusercontent.com/372735/54678548-a7952580-4b05-11e9-8d3a-cac9da089e79.png">
<img width="486" alt="Screenshot 2019-03-20 at 11 44 49" src="https://user-images.githubusercontent.com/372735/54678546-a6fc8f00-4b05-11e9-9671-c66d176f60ac.png">

jankeromnes commented:
Shouldn't `yarn` automatically run `npm install`?

While trying to open this PR in Gitpod, it ran `yarn` in pre-build, but `yarn dev --host 0.0.0.0` fails with:

> error There was a problem loading the local develop command. Gatsby may not be installed. Perhaps you need to run "npm install"?
> Error: Cannot find module 'toidentifier'

However, if I manually run `npm install` (takes a long time) then afterward everything works.

Potentially related, a bunch of files are considered "Deleted" while I haven't done anything in that workspace yet:

<img width="243" alt="Screenshot 2019-03-20 at 14 48 10" src="https://user-images.githubusercontent.com/599268/54689236-66117400-4b1f-11e9-8298-8fdfde57cdaa.png">

jankeromnes commented:
- [x] rebased
- [x] hide all background logos on mobile
- [x] fixed off-by-one error in responsive breakpoints
- [x] added docs topic selector

Firefox:

<img width="320" alt="Screenshot 2019-03-20 at 18 52 11" src="https://user-images.githubusercontent.com/599268/54707474-7afeff00-4b41-11e9-83e2-9172a0a27ea8.png">

Chrome:

<img width="320" alt="Screenshot 2019-03-20 at 18 53 14" src="https://user-images.githubusercontent.com/599268/54707491-83efd080-4b41-11e9-9d49-05258189a405.png">

svenefftinge commented:
Cool! Is it on purpose that the combo box only contains the first level of headings?

jankeromnes commented:
Yes (top pages then have sub-links) but I'm happy to add all entries if you think that's better.

svenefftinge commented:
With a little indentation I think it would be simpler. Also there is still something in the main menu.

jankeromnes commented:
> With a little indentation I think it would be simpler.

Sure, will try.

> Also there is still something in the main menu.

You want the expandable docs tree removed from the main menu? (I thought it was somewhat useful yet discreet enough to not be a problem.)

jankeromnes commented:
<img width="381" alt="Screenshot 2019-03-21 at 09 27 16" src="https://user-images.githubusercontent.com/599268/54740424-a6becb00-4bbb-11e9-84db-11f35d14f7d1.png">

svenefftinge commented on src/components/Header.tsx:
```diff
@@ -115,38 +130,69 @@ interface HeaderProps {
                     <HomepageLink to="/features">Features</HomepageLink>
                     <HomepageLink to="/pricing">Pricing</HomepageLink>
-                    <HomepageLink to="/docs">Docs</HomepageLink>
+                    <div>
```
I think this can be removed in favor of the combo box.

jankeromnes replied:
@ChristinFrohne agrees that it's confusing to have the menu twice, so I'll remove it. Thanks for the feedback!

SharifPoetra commented:
The editor on mobile chrome very hard to control!
